packages/core/test/actions.test.ts

```diff
@@ -522,6 +562,28 @@ describe('entry/exit actions', () => {
       ).toEqual(['exit_walk', 'exit_red', 'enter_red', 'enter_walk']);
     });
 
+    it('should exit current node and enter target node when target is not a descendent or ancestor of current', () => {
+      expect(
+        deepMachine2
+          .transition('A.A1', 'CHANGE_TO_A2A')
+          .actions.map((a) => a.type)
+      ).toEqual(['exit_A1', 'enter_A2', 'enter_A2a']);
+    });
+
+    it('should exit current node and enter target node when target is ancestor of current', () => {
+      expect(
+        deepMachine2
+          .transition('A.A2.A2a', 'CHANGE_TO_A')
+          .actions.map((a) => a.type)
+      ).toEqual(['exit_A2a', 'enter_A', 'enter_default']);
+    });
+
+    it('should enter all descendents when target is a descendent of current', () => {
+      expect(
+        deepMachine2.transition('A', 'CHANGE_TO_A2').actions.map((a) => a.type)
+      ).toEqual(['enter_A', 'enter_A2', 'enter_A2a']);
+    });
```

Comment:
You have mentioned in [the other comment](https://github.com/statelyai/xstate/pull/3623#discussion_r984953334) that you are unsure about this test and you have mentioned something about exit actions there. Could you elaborate? I don't see exit actions anyhow related to this transition and I'm unsure what you have observed.

Reply:
I was referring to this from the documentation:

"External transitions will exit and reenter the state node in which the transition is defined. "

That made me wonder if there was supposed to be an exit from A in this case. Maybe I'm misinterpreting it.

Reply:
I would have to recheck to be double sure but I think that yes, it should be exited for the external transition. The [transition domain](https://www.w3.org/TR/scxml/#getTransitionDomain) is the common ancestor of the source and the target and all states below the transition domain are *exited* (some might also get reentered).

This particular test had no exit action on the `A` - and that's why I wasn't sure what kind of an effect you are observing here.

Reply:
Oh, you're right. No wonder. I have updated it to include the exits so it's clear it's doing the right thing.